src/App.tsx

```diff
@@ -17,18 +44,19 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter query={query} setQuery={setQuery} selectedValue={selectedValue} setSelectedValue={setSelectedValue} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0
+                ? <Loader />
+                : <TodoList todos={filteredTodos} selectedTodo={selectedTodo} setSelectedTodo={setSelectedTodo} />}
```

Comment:
let's add isLoading and if after filter filtered todos length is 0 show empty message